### PR TITLE
Allow to install toml11 library with CMake.

### PR DESCRIPTION
**Pull request by KerstinKeller**
This PR allows to install toml11 library with CMake.
E.g. if you do the following:

```
mkdir _build
cd _build
cmake ..
cmake --build .
cmake --build . --target=install
```

you can install the header files and by CMake generated config files into your system.
You can also use the project from CMake like this

```
find_package(toml11 REQUIRED)

add_executable(my_exe test.cpp)
target_link_libraries(my_exe PRIVATE toml11:toml11)
```

Also added option to not build test, to pass `-Dtoml11_BUILD_TEST=OFF`.

### CODE REVIEW

**Comment by ToruNiina** [comment]
Thank you for PR!
Basically, it looks good to me. But I found one thing that might not be okay, so I added a comment.
Could you take a look when you have time?

**Comment by KerstinKeller** [comment]
Sure, but I cannot see any comments on the commit? Can you make it again please, or post it here?

**Comment by ToruNiina** [comment]
I'm sorry I didn't realize it hasn't done yet. Can you see it now?

File: CMakeLists.txt [hunk 1]
```diff
@@ -34,5 +44,62 @@ else()
...
+)
+install(EXPORT toml11Targets 
+  FILE toml11Targets.cmake
+  DESTINATION ${toml11_install_cmake_dir} 
+  NAMESPACE toml11::
+)
+
+if (toml11_BUILD_TEST)
 add_subdirectory(tests)
+endif ()
```
[the remark this answers is not in the document]
Comment (KerstinKeller) [reply CMakeLists.txt]: Ah ok, I never realized that / had any problems with it, but I can certainly add the newline at the end of the CMakeLists.txt files.

[the remark this answers is not in the document]
Comment (ToruNiina) [reply CMakeLists.txt]: Thank you!

**Pull request merged**